sw/device/silicon_creator/mask_rom/mask_rom.c

```diff
@@ -177,10 +173,12 @@ void mask_rom_boot(void) {
     }
   }
 
-  // Boot failed for all ROM_EXTs allowed by boot policy
-  // boot_failed(boot_policy); // Boot Policy Module
-  asm volatile("unimp");
+  return kErrorMaskRomBootFailed;
+}
+
+void mask_rom_main(void) {
+  rom_error_t error = mask_rom_boot();
   while (true) {
-    wait_for_interrupt();
+    shutdown_finalize(error);
```

Comment:
Since this should be `noreturn`, the `while` is unnecessary.

Reply:
My thoughts here were if you manage to glitch out of the `wfi` maze in shutdown_finalize and return, we get a whole new chance to re-enter finalize and kill the chip again.

@moidx 
I'm not sure how realistic this is and whether we'd be better of just marking this noreturn.

Reply:
Once entering the `wfi` maze, I think there is no path to return to this function, is that correct?

Reply:
There _should_ be no exit from the `wfi` maze.  If we accept that, then this function is `noreturn`.